src/main/java/net/dries007/tfc/objects/te/TEFirePit.java

```diff
@@ -581,6 +583,11 @@ public void onConvertToGrill(EntityPlayer player, ItemStack stack)
         attachedItemStack = stack.splitStack(1);
     }
 
+    public void onRemoveAttachment()
+    {
+        attachedItemStack = ItemStack.EMPTY;
+    }
+
```

Comment:
No, this should handle dropping and giving back to player, as the attachedItemStack needs to be given back to the player exactly.